doc/manuals/orion-api.md

```diff
@@ -2295,6 +2326,676 @@ Some considerations to take into account when using custom notifications:
+c|log
+```
+
+results in
+
+```
+1.144222799920162
+```
+
+#### log10
+
+Return the base 10 logarithm of a given number
+
+Extra arguments: none.
+
+Example (being context `{"c": 3.14}`):
+
+```
+c|log10
+```
+
+results in
+
+```
+0.49692964807321494
+```
+
+#### log2
+
+Return the base 2 logarithm of a given number
+
+Extra arguments: none.
+
+Example (being context `{"c": 3.14}`):
+
+```
+c|log2
+```
+
+results in
+
+```
+1.6507645591169025
+```
+
+#### sqrt
+
+Return the square root of a given number
+
+Extra arguments: none.
+
+Example (being context `{"c": 3.14}`):
+
+```
+c|sqrt
+```
+
+results in
+
+```
+1.772004514666935
+```
+
+#### replaceStr
+
+Replace occurrences of a string with another in the input string.
+
+Extra arguments:
+* Source string to replace
+* Destination string to replace
+
+Example (being context `{"c": "foobar"}`):
+
+```
+c|replaceStr('o','u')
+```
+
+results in
+
+```
+"fuubar"
+```
+
+#### mapper
+
+Returns a value among several choices based in one to one mapping. This function is based in an array of *values* and an array of *choices* (which length is exactly the same). Thus, if the input value is equal to the *i*-th item of *values*, then *i*-th item of *choices* is returned.
+
+This transformation returns `null` if some problem with the arguments is found (i.e. input is not found among the values, choices length is not exacly the same as values, the input is not an string, etc.)
+
+Extra arguments:
+* values array
+* choices array
+
+Example (being context `{"c": "fr", "values": ["es", "fr", "de"], "choices": ["Spain", "France", "Germany"]}`):
+
+```
+c|mapper(values,choices)
+```
+
+results in
+
+```
+"France"
+```
+
+#### thMapper
+
+Returns a value among several choices based in threshold values. This function is based in an array of *values* and an array of *choices* (which length is exactly the same as values plus one). Thus, if the input value is between the *i*-th and the *i+1*-th item of *values*, then *i*+1-th item of *choices* is returned.
+
+This transformation returns `null` if some problem with the arguments is found (i.e. choices length is not exacly the same as values plus one, some of the items in the values array is not a number, etc.)
+
+Extra arguments:
+* values array
+* choices array
+
+Example (being context `{"c": 0.5, "values": [-1, 1], "choices": ["low", "medium", "high"]}`):
+
+```
+c|thMapper(values,choices)
+```
+
+results in
+
+```
+"medium"
+```
+
+### Failsafe cases
+
+As failsafe behaviour, evaluation returns `null` in the following cases:
+
+* Some of the transformation used in the expression is unknown
+* Operations with identifiers that are not defined in the context are used. For instance, `(A==null)?-1:A` will result in `null` (and not `-1`) if `A` is not in the context, due to `==` is an operation that cannot be done on undefined identifiers. However, `A||-1` will work (i.e. `-1` will result if `A` is not in the context), as `||` is not considered an operation on `A`.
```

Comment:
```suggestion
* Operations with identifiers that are not defined in the context are used. For instance, `(A==null)?1:A` will result in `null` (and not `1`) if `A` is not in the context, due to `==` is an operation that cannot be done on undefined identifiers. However, `A||1` will work (i.e. `1` will result if `A` is not in the context), as `||` is not considered an operation on `A`.
```

As `A||-1` include minus operator and it is a "Known limitation", I suggest using this example

Reply:
```suggestion
* Operations with identifiers that are not defined in the context are used. For instance, `(A==null)?0:A` will result in `null` (and not `0`) if `A` is not in the context, due to `==` is an operation that cannot be done on undefined identifiers. However, `A||0` will work (i.e. `0` will result if `A` is not in the context), as `||` is not considered an operation on `A`.
```

Reply:
Fixed in [fbde34e](https://github.com/telefonicaid/fiware-orion/pull/4511/commits/fbde34e73bc50a01e63a11b9473a295a7723262a)